src/client/main/macroTarget.factory.js

```diff
@@ -0,0 +1,159 @@
+/* global _ */
+(function() {
```

Comment:
As long as we agree to go full-OOP in Angular 2 and follow the camel case standard, I'll survive.